extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-story-quiz-1.0.css';
+import {createShadowRootWithStyle} from './utils';
+import {dev} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {toArray} from '../../../src/types';
+
+/** @const {!Array<string>} */
+const answerChoiceOptions = ['A', 'B', 'C', 'D'];
+
+/** @const {string} */
+const TAG = 'amp-story-quiz';
+
+/**
+ * Generates the template for the quiz
+ * @param {HTMLElement} element
+ * @return {HTMLElement}
+ */
+const buildQuizTemplate = element => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amp-story-quiz-container">
+      <div class="i-amp-story-quiz-prompt-container"></div>
+      <div class="i-amp-story-quiz-option-container"></div>
+    </div>
+  `;
+};
+
+/**
+ * Generates the template for each option
+ *
+ * @param {HTMLElement} option
+ * @return {HTMLElement}
+ */
+const buildOptionTemplate = option => {
+  const html = htmlFor(option);
+  return html`
+    <span class="i-amp-story-quiz-option">
+      <span class="i-amp-story-quiz-answer-choice"></span>
+    </span>
+  `;
+};
+
+export class AmpStoryQuiz extends AMP.BaseElement {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private {boolean} */
+    this.hasReceivedResponse_ = false;
+
+    /** @private {?ShadowRoot|HTMLDivElement} */
+    this.quizElt_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.quizElt_ = buildQuizTemplate(this.element);
+    this.attachContent_();
+    this.attachOptionActionHandlers_();
+    createShadowRootWithStyle(this.element, this.quizElt_, CSS);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
```

Comment:
Given that publishers have no control over the size of the element, is this still relevant?

Reply:
Changed supported layout type to 'flex-item' only